pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -29,6 +30,31 @@ import (
 	"istio.io/istio/pilot/pkg/networking/util"
 )
 
+// BuildRoutes produces a list of listeners and referenced clusters for all proxies
+func (configgen *ConfigGeneratorImpl) BuildRoutes(env model.Environment, node model.Proxy, routeName string) (*xdsapi.RouteConfiguration, error) {
+	// TODO: Move all this out
+	proxyInstances, err := env.GetProxyServiceInstances(&node)
+	if err != nil {
+		return nil, err
+	}
+
+	services, err := env.Services()
+	if err != nil {
+		return nil, err
+	}
+
+	// ensure services are ordered to simplify generation logic
+	sort.Slice(services, func(i, j int) bool { return services[i].Hostname < services[j].Hostname })
```

Comment:
AFAICT the only way we use the `services` list is to create a `map[hostname]service`, which by design has random iteration ordering; I don't think we get any value out of sorting here.